File: solutions/cpp/grade-school/1/grade_school.cpp
#include "grade_school.h"
#include <algorithm>
#include <string>
#include <utility>
#include <vector>

namespace grade_school {
void school::add(std::string nama, int kelas) {
  std::vector<std::string> tmp = murid[kelas];
  tmp.push_back(nama);
  std::sort(tmp.begin(), tmp.end());
  murid[kelas] = tmp;
}

std::map<int, std::vector<std::string>> school::roster() const { return murid; }

std::vector<std::string> school::grade(int kelas) const {

  std::vector<std::string> tmp{};
  if (murid.count(kelas) <= 0)
    return tmp;
  tmp = murid.at(kelas);
  return tmp;
}
} // namespace grade_school

File: solutions/lua/grade-school/1/grade-school.lua
local School = {}

function School:roster()
  return self.student
end

function School:add(name, class)
  if self.student[class] == nil then
    self.student[class] = {
      name
    }
  else
    table.insert(self.student[class], name)
    table.sort(self.student[class])
  end
end

function School:grade(index)
  if self.student[index] == nil then
    return {}
  end
  return self.student[index]
end

function School:new()
  local student = {}
  local n = {
    roster = School.roster,
    add = School.add,
    student = student,
    grade = School.grade
  }
  setmetatable(n, School)
  return n
end

return School
